packages/grid_client/tests/modules/gateways.test.ts

```diff
@@ -190,15 +192,34 @@ test("TC1237 - Gateways: Expose a VM Over Gateway", async () => {
   expect(gatewayResult[0].backends).toStrictEqual(backends);
 
   const domain = "https://" + gatewayResult[0].domain;
+  let reachable = false;
+
+  for (let i = 0; i < 15; i++) {
+    fetch(domain, { mode: "no-cors" })
+      .then(r => {
+        log("gateway is reachable");
+        reachable = true;
+        i = 15;
+      })
+      .catch(e => {
+        log("gateway is not reachable");
+      });
+    const wait = await setTimeout(10000, "Waiting for gateway to be ready");
+    log(wait);
+  }
 
-  //Verify that the domain points to VM and lists the directories available in the VM
-  exec("curl " + domain, function (_, stdout) {
-    log(stdout);
-    expect(stdout).toContain("Directory listing for /");
-    expect(stdout).toContain("bin/");
-    expect(stdout).toContain("dev/");
-    expect(stdout).toContain("etc/");
-  });
+  if (reachable) {
+    //Verify that the domain points to VM and lists the directories available in the VM
+    exec("curl " + domain, function (_, stdout) {
```

Comment:
why using `exec` to `curl` http request? can't we use `fetch` or `axios` to do it?

Reply:
i think `fetch` also works but i encountered a case were `fetch` returned a false positive.
The case happened with node 11 as sometimes `fetch` will return that it's reachable but the gateway will still give `404 not found` for a couple of minuets so the extra `curl` command confirms that the gateway is actually working and asserts that the desired result appeared.

Reply:
done in https://github.com/threefoldtech/tfgrid-sdk-ts/pull/107/commits/de767e12e94a87712fd099f88828ca329928ff9d instead of checking with `curl` now the check is done with `fetch` and the assertion is done for the status and status code.

Reply:
we can still check the content of the response

Reply:
Re-added in https://github.com/threefoldtech/tfgrid-sdk-ts/pull/107/commits/67c62b4a0910f8c9e12f83a048e13bac6b637f79 using fetch